### DISCUSSION TITLE
Failed to connect to rs-ny.rustdesk.com:21116

### DISCUSSION THREAD

**Discussion by kenichi512** [rustdesk/rustdesk#5538] [Q&A] [answered]
### Bug Description

I'm having trouble connecting from a desktop PC 1 (Ethernet) to another desktop PC 2 (Wi-Fi). I should clarify that the computers are in different places and connected to different networks or ISPs. I had no problems before but recently from PC 1 I have not been able to connect to PC 2. Last week I was getting an error message saying "deadline has elapsed" and now I am getting an error message "Failed to connect to rs-ny.rustdesk. com:21116". While on PC 1 it says "Not ready. Check your internet connection", on PC 2 it says "Ready". When I try to connect from PC 2 to PC 1, Rustdesk on PC 2 says that the remote computer is not online despite being connected to the internet, and while I have problems on PC 1, from an Android phone (connected to the same network as PC 1) I can connect to PC 2. What can be wrong on PC 1? The Rustdesk version currently installed on PC 1 is 1.2.2 (I reinstalled thinking that this would fix the problem) but previously I was using version 1.1.9 (same as installed on PC 2). 

### How to Reproduce

Installed

### Expected Behavior

It should get connected

### Operating system(s) on local side and remote side

Windows 10 -> Windows 10

### RustDesk Version(s) on local side and remote side

1.2.2 (nightly) -> 1.1.9

### Screenshots

![Rustdesk error PC1](https://github.com/rustdesk/rustdesk/assets/25917998/2cc36686-ca43-41d0-aebc-cacd198d1c08)

### Additional Context

This is the log of PC 1 (I didn't know how to get it until I found the path in another thread)

`
[2023-08-28 11:06:19.836085 -05:00] DEBUG [libs\hbb_common\src\config.rs:434] Configuration path: C:\Users\namehide\AppData\Roaming\RustDesk\config\RustDesk_default.toml
[2023-08-28 11:06:37.839480 -05:00] DEBUG [src\rendezvous_mediator.rs:594] Failed to connect to rs-ny.rustdesk.com:21115
[2023-08-28 11:06:37.839539 -05:00] DEBUG [src\rendezvous_mediator.rs:599] query onlines timeout 3s
[2023-08-28 11:06:37.872447 -05:00] DEBUG [src\rendezvous_mediator.rs:594] Failed to connect to rs-ny.rustdesk.com:21115
[2023-08-28 11:06:37.872509 -05:00] DEBUG [src\rendezvous_mediator.rs:599] query onlines timeout 3s
[2023-08-28 11:06:37.872447 -05:00] DEBUG [src\rendezvous_mediator.rs:594] Failed to connect to rs-ny.rustdesk.com:21115
[2023-08-28 11:06:37.872567 -05:00] DEBUG [src\rendezvous_mediator.rs:599] query onlines timeout 3s
[2023-08-28 11:06:37.874447 -05:00] DEBUG [src\rendezvous_mediator.rs:594] Failed to connect to rs-ny.rustdesk.com:21115
[2023-08-28 11:06:37.874503 -05:00] DEBUG [src\rendezvous_mediator.rs:599] query onlines timeout 3s
[2023-08-28 11:06:37.884051 -05:00] DEBUG [libs\hbb_common\src\config.rs:434] Configuration path: C:\Users\namehide\AppData\Roaming\RustDesk\config\RustDesk_default.toml
[2023-08-28 11:06:37.885508 -05:00] DEBUG [libs\hbb_common\src\config.rs:434] Configuration path: C:\Users\namehide\AppData\Roaming\RustDesk\config\RustDesk_default.toml
[2023-08-28 11:06:37.905868 -05:00] DEBUG [libs\hbb_common\src\config.rs:434] Configuration path: C:\Users\namehide\AppData\Roaming\RustDesk\config\RustDesk_hwcodec.toml
[2023-08-28 11:06:37.921806 -05:00] INFO [src\flutter.rs:794] Session 521327037 start, render by flutter texture rgba plugin
[2023-08-28 11:06:37.935056 -05:00] INFO [libs\scrap\src\common\codec.rs:549] cpu num:4, cpu usage:None, codec thread:1
[2023-08-28 11:06:37.935819 -05:00] DEBUG [libs\hbb_common\src\config.rs:434] Configuration path: C:\Users\namehide\AppData\Roaming\RustDesk\config\RustDesk_hwcodec.toml
[2023-08-28 11:06:37.958772 -05:00] DEBUG [libs\hbb_common\src\config.rs:434] Configuration path: C:\Users\namehide\AppData\Roaming\RustDesk\config\RustDesk_hwcodec.toml
[2023-08-28 11:06:49.621662 -05:00] DEBUG [libs\hbb_common\src\config.rs:434] Configuration path: C:\Users\namehide\AppData\Roaming\RustDesk\config\RustDesk_default.toml
[2023-08-28 11:06:55.842863 -05:00] DEBUG [src\rendezvous_mediator.rs:594] Failed to connect to rs-ny.rustdesk.com:21115
[2023-08-28 11:06:55.842947 -05:00] DEBUG [src\rendezvous_mediator.rs:599] query onlines timeout 3s
[2023-08-28 11:06:55.985957 -05:00] INFO [src\client.rs:280] rendezvous server: rs-ny.rustdesk.com:21116
[2023-08-28 11:06:55.986120 -05:00] ERROR [src\client.rs:2409] Connection closed: Failed to connect to rs-ny.rustdesk.com:21116: Please try later
[2023-08-28 11:06:55.986177 -05:00] INFO [src\client\io_loop.rs:853] sync transfer job status
[2023-08-28 11:06:55.986654 -05:00] DEBUG [libs\hbb_common\src\config.rs:434] Configuration path: C:\Users\namehide\AppData\Roaming\RustDesk\config\RustDesk_default.toml
[2023-08-28 11:06:55.986835 -05:00] INFO [src\client\io_loop.rs:864] meta: TransferSerde { write_jobs: [], read_jobs: [] }
[2023-08-28 11:06:55.986883 -05:00] INFO [src\client.rs:1856] Video decoder loop exits
[2023-08-28 11:06:55.986887 -05:00] INFO [src\client.rs:1884] Audio decoder loop exits
[2023-08-28 11:06:56.012415 -05:00] DEBUG [libs\hbb_common\src\config.rs:434] Configuration path: C:\Users\namehide\AppData\Roaming\RustDesk\config\RustDesk_hwcodec.toml
[2023-08-28 11:06:58.230957 -05:00] DEBUG [src\rendezvous_mediator.rs:594] Failed to connect to rs-ny.rustdesk.com:21115
[2023-08-28 11:06:58.231018 -05:00] DEBUG [src\rendezvous_mediator.rs:599] query onlines timeout 3s
`

**Comment by xlionjuan** [comment]
Build your own server instead.

**Reply by Manxtu** [reply]
Id love to build my own server on a raspberry pi and I have tried but unsuccessfully - Can anyone out there help?

**Reply by playbackandrewind** [reply]
Get a free docker container https://aws.amazon.com/en/free/containers/

then install your own server:

https://rustdesk.com/docs/en/self-host/rustdesk-server-oss/docker/

**Reply by xlionjuan** [reply]
> Id love to build my own server on a raspberry pi and I have tried but unsuccessfully - Can anyone out there help? 

First, learn how to install docker on your system, if you're using Ubuntu, it is convenient to install it though snap.
```bash
sudo snap install docker
```
Feel free to asking more server or Docler problems.

**Reply by xumixu** [reply]
Seems like AWS servers have limited hours for free. AFAIK 1 year has more than the free 750 hours.

**Reply by playbackandrewind** [reply]
750 hours a MONTH, and one year for free.

And yes, it works flawless - at this moment - while rs-ny.rustdesk.com is down.

Point ID-Server and Relay-Server in the network settings to your AWS host on ALL MACHINES running rustdesk, and you're back online.

**Reply by xumixu** [reply]
I could have swear it didn't say monthly.

Ended up setting it in up in a windows PC + port forwarding + dynamic dns.

**Comment by ascendforever** [comment]
I am also getting this error "Failed to connect to rs-ny.rustdesk. com:21116" today, Jan 31st, 2024. Rustdesk worked perfectly 12 hours ago, and now is non-functional. I have tested on versions 1.2.3-1 and 1.2.3-3 and both have the issue.

**Comment by RDG58** [comment]
I'm having exactly the same problem - worked fine a few hours ago and now get "Failed to connect to rs-ny.rustdesk.com:21116: Please try later". I switched to the newest version because the prior installation also stopped working. Can anyone suggest an alternative service?

**Reply by xiadaren** [reply]
我今天刚玩这个，我是类似情况，是不是还要搞什么ID服务器API之类的

**Reply by xlionjuan** [reply]
@xiadaren HBBS(ID)  HBBR(Relay 中繼)
就這樣 let's it

@RDG58 host your own server instead to avoid this problem.

**Comment by cyber-sec0** [comment]
same here

**Comment by OpenSoftsrl** [comment]
i have the same problem.

**Comment by TheApostleHunter** [comment]
I'm hoping this gets resolved shortly, it's affecting quite a few of us.

**Comment by rc621** [comment]
same here... hope they will fix it soon - thanks

**Comment by Harry-Sanchez** [comment]
yo igual pense que era mi pc pero por lo que veo es el servidor. el problema lleva mas menos 12 horas.
alguien sabe cuanto tiempo mas le daran solucion.????

**Reply by xlionjuan** [reply]
They still being DDoS

**Comment by solerzhuxy** [comment]
same here！

**Comment by H-tr** [comment]
same here!

**Comment by aodix85** [comment]
Yup, same here....I have been planning on building a server for my connections out for a while now, maybe this is the time to do that...but in the meantime hopefully this gets sorted soon

**Reply by xumixu** [reply]
Yep, downloaded teamviewer now to survive while i learn and try to set a personal server


Ended up setting it up in windows. Wasted hours following the self host guide. Then being curious executed the setup executable (which was never mentioned in the guide) and it set itself up in 5 minutes... Only had to add the public key manually at the end.

**Reply by kozman1000** [reply]
5 minutes and it was up and running? What version of the server did you use? 1.1.10-3?

**Reply by xumixu** [reply]
More or less. I went to the download site mentioned in the guide and downloaded the last package for windows. Then executed the "RustDeskServer setup.exe".

The installer version is 1.1.10.0 (from details tab), the rar didn't have the version on the filename.

You need to have the PC visible from the internet, that means static IP (or dynamic DMS) and portforwarding, that you have to do by yourself (I did it previously for Jellyfin).

**Reply by kozman1000** [reply]
I've done setups with NoIP and DynDNS + port forwarding so it shouldn't be too bad. I'm wanting to do a Linux setup but Windows is more available at my org. I'll have to try with both and see which works better for our needs.

**Comment by zbxyu83** [comment]
same here!

**Comment by owl-city** [comment]
Me too!

**Comment by EZElkins** [comment]
Using direct IP access can solve this problem temporarily.

**Reply by ThreeMinutesFix** [reply]
how?

**Reply by EZElkins** [reply]
Target RustDesk: Settings - Secuity - Enable Direct IP Access

**Reply by playbackandrewind** [reply]
Without a working ID-Server, this will not work.

**Reply by EZElkins** [reply]
You need to input IP in Control Remote Desktop input box.

**Reply by owl-city** [reply]
I think this man has no idea how a server works.

**Reply by xumixu** [reply]
Thanks, It works for local connections and for computers than can be exposed to internet through port redirection (i use brynhildr for that though). But for the computers at my lab, no luck :(

**Reply by kozman1000** [reply]
No, EZElkins is right. If a machine is on the >same network< as your PC, type in it's IP into the Control Remote Desktop field and it will connect.

**Reply by xlionjuan** [reply]
@EZElkins is right, @playbackandrewind  need to do some research on this software.

**Comment by Ahirhb** [comment]
i am facing same issue

**Comment by webtomek** [comment]
Same error here. I'm working from home and just spent an hour trying to fix this. Now I'm back to Teamviewer and other stuff...

**Comment by omarabdelwahabmb** [comment]
Failed to connect to rs-ny.rustdesk.com:21115

**Comment by Eduart7** [comment]
Same here!

**Comment by ThreeMinutesFix** [comment]
yes its not working.

**Comment by Jader2000** [comment]
Yup. Not working.

**Comment by jmperezfminsait** [comment]
Are there any news @rustdesk ?
I am having the same error.

**Comment by playbackandrewind** [comment]
PSA - RUSTDESK IS DOWN: **DDOS on Rustdesk Public Servers** => Non-Selfhosted Rustdesk does NOT work currently #7027 

https://github.com/rustdesk/rustdesk/issues/7027

So please close this issue.

**Reply by gmhall506** [reply]
But I'm self-hosted and mine is down too.

**Reply by playbackandrewind** [reply]
Works flawless here - all client systems are up and running flawless with private server.

Point ID-Server AND Relay-Server to your own private server.

**Comment by Satalink** [comment]
I'm having the same issue.  If this is related to an AWS server reaching its max bandwidth for the month, this is a serious issue as a viable product.   If this is causing users to abandon RustDesk in favor of TeamViewer, it's also costing "brand recognition".   Ultimately, it's not good for the RustDesk product.  I love RustDesk and look forward to it coming back online soon.

**Reply by playbackandrewind** [reply]
The public server is hosted by vultr.com - not AWS. And It's a DDoS Attack, nothing else.

See -> https://github.com/rustdesk/uptime/issues

**Reply by Satalink** [reply]
I suggest using CLoudFlare DNS with Proxy enabled (even the free version offers decent protection against DDoS)

**Comment by Mingggzzzx** [comment]
Thought it was an issue with my connectivity, apparently its not after a quick search.

**Comment by Jack-InGitHub** [comment]
Same here!

**Comment by Satalink** [comment]
# I found a way to make RustDesk work! #
 *** (no relay server required) ***

### RustDesk Host ### 
*[Settings] -> Security ->  (Unlock Security Settings)*
 ==>  **Enable Direct IP Access**
 ==>  Port:   **xxxxx**

### LAN Router ###
Go to your router and configure port forwarding using your [xxxxx] port to your RustDesk host.
( *Method depends on your router* )
==>   ***[xxxxx] => {LAN ip of host}***  

### RustDesk Client ###
- On your RustDesk Client in the field where you normally put the RustDesk ID 
*Control Remote Desktop*
==>  **{ipaddress}:xxxxx** or **{domain.tld}:xxxxx**  
( domain if you use NoIP or  have static IP DNS configured on a domain )

**Reply by playbackandrewind** [reply]
This works -like a Microsoft remote desktop-only between two machines. If you want to use more than two machines, set up your own private server -> https://rustdesk.com/docs/en/self-host/

**Reply by xumixu** [reply]
And if using a windows machine, skip the guide and just use the rustdeck server installer. It does everything by itself. Just need to portforward your server to be accessed for the internet, and add it to the clients.
Also when already running the server, it generates a public key hbbs key in the installation folder, which is not mentioned in the guide and it can be used on clients to encrypt the connection (unencrypted by default)

**Reply by orangeudx** [reply]
might as well use vnc then :-/

**Comment by Wuffimaus** [comment]
Same Problem here 🙁

**Comment by kozman1000** [comment]
Uh oh, 9:14PM CST...I see it's up?

**Comment by rketsawas** [comment]
Oh ...It's back!?

**Comment by kozman1000** [comment]
Back is probably too strong a word to use. It's slow so probably still getting hammered but didn't error out on rs-ny.rustdesk.com so maybe it's trying to come online again. Have to wait and see.

**Comment by Satalink** [comment]
Final note from me.

For my needs, the direct ip connect setup is all I need.  I suggest the RustDesk team make this option main stream in the initial configuration and use the VPS server (AWS or whatever it is) a more advanced configuration.  Basically, the opposite of what is coded in the UI now.   The most important thing is that you don't want to suffocate all of the outstanding work that has been done on this project with unreliability issues.

If you insist on using the VPS method as a default, you should add a load balancer and add an auto scalable VPS service.  Not free, but if configured correctly it wouldn't be very expensive.  The direct connect configuration (mentioned above) would greatly reduce much of the normal traffic and the scalable VPS on a load balancer would afford you stability for those that need connections to more than single direct connects.  Although, there's nothing stopping someone from port mapping multiple machines with the direct connect method.   A comment from someone earlier about that is confusing to me, but probably some level of ignorance on my behalf.

**Reply by kozman1000** [reply]
It's easy to say what someone else should do with their services when it is them and not you paying for the hosting, etc. DDoS at any scale can take a server down. The kind of resources required to have resiliency against what is happening presently requires a lot of $$ and likely being on colocated infrastructure and 100Gb connections so as not to have the connection saturated. That costs a LOT of $$ and is probably outside the financial reach of the Rustdesk org. The mere monthly cost of hosting 1 server to do this is very high. Imagine if they wanted to host 10 server to have resiliency? Suddenly their costs, not ours, increases 10 fold.

I would love to see what someone suggested in terms of non-centralized traffic relaying a la Torrent network whereby so long as someone is leaving their client+home Internet connection running they are basically acting as a local relay so even if the central relay goes down it doesn't take out the whole network with it. Mitigation against a single point of failure takes a lot of thought and thorough planning but sometimes it just cannot be helped. Bad actors are not going to stop acting bad whether they are motivated by $ or just plain malice.

Having Direct IP connection on a local network enabled by default (going forward) in the client should be, as a consequence of what is happening presently, a default setting that is enabled so at least it is something that will work for a user in the event that a public relay goes down or their own private server goes down. They won't be totally dead in the water. Everyone here has an opinion on the matter because hindsight is 20/20. The Rustdesk devs have a lot to think about and will have to pivot in some areas as the now see, sadly, someone has it out for them.

In an ideal world, I'd love to see some big datacenter players step up and donate a server or two and some bandwidth. Chances of that are probably nil. I'd rather see a GoFundMe, Paypal or Patron-type outreach happen to fund standing up some more servers that can spread and balance the load. Of course, that doesn't mean those servers won't be targets as well. It will be an endless game of cat and mouse because whoever is attacking will just scale up as well.

**Reply by Satalink** [reply]
I'll donate my time to assist in a bullet proof system solution.  Just reach out.

**Reply by Satalink** [reply]
My first recommendation would be to migrate NameServer to CloudFlare (free account)

 current rustdesk.com NS on GoDaddy ( notoriously horrible )
 `ns41.domaincontrol.com`
 `ns42.domaincontrol.com`
 
 CloudFlare offers extensive utilities and tools to combat bad actors.  (even with the free account)!!!
 
 Once you have your NS reconfigured to CloudFlare, you can assess the DDoS much easier with their dashboard traffic analysis tools.
 
 If in the event you need to upgrade, you will have even more tools afforded to you to deal with DDoS.  If after that, you can move to create a LoadBalancer within CloudFlare.  
 
 As for the VPS servers,  If you end up needing to configure a CF LoadBalancer, you can move to creating a half dozen AWS (free tier) accounts and configure the CF LoadBalancer to use each of them.   
 
 Initially, just moving the NS (nameserver) from CloudFlare's default NS servers to a free CF account would do a lot to combat the DDoS.   The other recommendations are steps to take in the event that doesn't negate the attack.
 
 I would start with that.  
 
 [CloudFlare Free Plan](https://www.cloudflare.com/plans/free/)
 
 [How to move your NS from GoDaddy to CloudFlare](https://support.artstorefronts.com/hc/en-us/articles/360033747991-How-To-Change-Your-Nameservers-from-GoDaddy-to-Cloudflare)

**Reply by xlionjuan** [reply]
Their ID server **isn't** based on HTTP protocol, so it can't use Cloudflare's DDoS protection or WAF for free as usual, though Cloudflare provide TCP DDoS protection, but it isn't cheap, and HBBS is mainly using UDP for heartbeat, and HBBS need to get client's real IP to initial TCP hole punching connections.

**Reply by Satalink** [reply]
That is a conundrum.   Maybe refactor HBBS to use http with RESTful to hand off the ip address?

**Reply by kozman1000** [reply]
If the bad actors are hammering just the IP, that's one thing. But if they're hammering the common Rustdesk TCP/UDP ports to death, then maybe code that dynamically throttles incoming connections requests based on heuristics that evaluate abnormal behavior (1000s of requests/sec, etc.) coming from a certain IP repeatedly and drops incoming packets from that IP until it stops or limits requests to one every 60 seconds. Just spitballing thought here. I know zero about how to code something like that but know that there are firewalls and appliances made the likes of Barracuda that have user configurable parameters that can do this. I've seen it in our own Barracuda appliance for incoming/outgoing email.

**Reply by Satalink** [reply]
@kozman1000 .. that's a good solution.   I've used ConfigServer csf firewall before. 

[CSF](https://configserver.com/configserver-security-and-firewall/)

**Reply by xlionjuan** [reply]
Very important to know, RustDesk's public server is for demonstration and charitable purpose, they shouldn't take much time for this.

For DDoS problem, I agree implement some firewall based on rate limit is a great idea, but don't know is it difficult to deploy on OS itself ?

**Reply by Satalink** [reply]
ConfigServer setup Ubuntu

https://www.digitalocean.com/community/tutorials/how-to-install-and-configure-config-server-firewall-csf-on-ubuntu

**Reply by kozman1000** [reply]
CSF would be good to help mitigation but bad actors would likely pump up their ingress traffic to the server and try to exhaust system resources. If CSF is constantly busy filtering traffic, CPU could get pegged at or near 100% just trying to protect you or your RAM would get chewed up because the queue gets huge. They already know most tactics an admin would do to try and stop them. That's why it's always better to tracert to the origin and reach out to downstream providers to cut off the source. Free IPs are rare as hens teeth so any ISP out there who's also hosting other customers on the same public IP where the attack is coming from would not want someone like the ICANN to get them kicked off the Internet. But, if all else fails: https://www.icann.org/en/blogs/details/how-to-report-a-ddos-attack-25-4-2013-en

A good defense also means having good offense as well.

**Comment by Satalink** [comment]
I can see how you got there.  

CSF does monitor traffic but once a bad actor has exceeded the threshold CSF blocks the address from the `nick interface` preventing further CPU consumption from that address.  You can configure it to be a temporary ban, permanent ban, or add rules that govern how many temp-bans are required over a definable period of time before it's moved into perm-ban.   

Read through these [feature bullets](https://configserver.com/configserver-security-and-firewall/).  Don't be intimidated by what all it can do,  just know that it can do exactly what you need it to do and there's more if you want to dive further later down the road.
[CSF Documentation](https://configserver.com/configserver-security-and-firewall/)

**Reply by kozman1000** [reply]
That does sound like a useful tool then if the Rustdesk guys were to run it on their Linux server. Every ounce of mitigation would help I would think. I'm thinking that when Ubuntu 24.04 comes out, I might try and set up my own server (even though I am primarily a Windows guy) and play with CSF. I play with Mint more than Ubuntu TBH and since Mint is Ubuntu based I would surmise CSF should work there as well. It does sound like a nifty tool. Thanks for the info about it.

**Comment by erzakiev** [comment]
does the problem still persist for everybody? I seem to still experience the issue..

**Comment by RDG58** [comment]
*It was working fine for me until about an hour ago and now will not
connect.*

**Comment by RDG58** [comment]
Once again, I'm getting the "Failed to connect to rs-ny.rustdesk.com:21116:
Please try later" error.

**Comment by jaydxr** [comment]
i have the same problem, do you fix it yet?

**Comment by oe3gwu** [comment]
I got the same problem. Not interested in hosting my own server at home.

**Comment by AlexCypri44** [comment]
I have the same problem as you for two days now.

**Comment by plivnik** [comment]
The same problem. Seems, that communication is blocked by  source address range(s). yesterday all nodes, connected via the same provider, was down, another provider was OK. Today all connections doesn't work
Mike
All my stations have private adrdessess, so without rs-ny.rustdesk.com I have no access.

**Comment by rustdesk** [comment, marked as answer]
https://www.reddit.com/r/rustdesk/comments/1cr8kfv/should_you_selfhost_a_rustdesk_server/
